### DISCUSSION TITLE
How to redraw edges?

### DISCUSSION THREAD

**Discussion by tpoxa** [bcakmakoglu/vue-flow#241] [Q&A] [answered]
Hey. 
I implement my custom node with handle rotation support. 
Handles are rotating, but edges stay in their old positions. Is there any way to ask vue-flow redraw edges to the current handle positions? 

Thanks for the vue-flow. It's a great project!

**Comment by bcakmakoglu** [comment, marked as answer]
To force-redraw edges you can use the `updateNodeInternals` action , which will re-calculate the handle positions and draw edges according to those new positions.
[More on that in the docs right here.](https://vueflow.dev/guide/node.html#dynamic-handle-positions-adding-handles-dynamically)

**Reply by tpoxa** [reply]
Thanks for the hint @bcakmakoglu. I tried that.  Unfortunately did not work. I am pretty sure I used that wrong. My rotation happens in pinia's action like this.

```js
async rotate() {
      let ids: string[] = []
      this.selectedNodes.forEach((el) => {
          el.data.spin++;
          el.data.handles.forEach((h) => {
            h.position =  (h.position_origin + el.data.spin) % 4
          })
      });
      this.elements.forEach((el) => {
        if (isNode(el)) {
          ids.push(el.id)
        }
      })
      await nextTick(async ()=> {
        updateNodeInternals(ids)
      })
    }
```

I guess there is no way I can emit that in the Custom Node, or can I?

Thank you for your help.

**Reply by tpoxa** [reply]
Ok, I managed to do that by watching props in a customNode and emitting `updateNodeInternals` from there. Thank you!
